Sources/Components/Components/Watchlist/WKWatchlistView.swift

```diff
@@ -38,7 +42,11 @@ private struct WKWatchlistContentView: View {
 							.padding([.top, .bottom], 6)
 							.frame(maxWidth: .infinity, alignment: .leading)
 						ForEach(section.items) { item in
-							WKWatchlistViewCell(itemViewModel: item, localizedStrings: viewModel.localizedStrings)
+							WKWatchlistViewCell(itemViewModel: item, localizedStrings: viewModel.localizedStrings, menuButtonDelegate: menuButtonDelegate)
+								.contentShape(Rectangle())
+								.onTapGesture {
+									delegate?.watchlistUserDidTapDiff(revisionID: item.revisionID, oldRevisionID: item.oldRevisionID)
```

Comment:
One thing that concerns me is that I see the "user did tap diff" printout when I'm trying to dismiss the menu while tapping outside the button. To me it feels weird, knowing that action will cause the user to navigate away to the diff view:

![out](https://github.com/wikimedia/wikipedia-ios-components/assets/3620196/87ffb7ee-fc8c-443d-b2ea-acd566e69a47)

I haven't had any luck yet figuring out how to fix it. A basic flag of whether or not a menu is displaying would suffice, and we can conditionally call this delegate method based on that, but I haven't had luck with that in SwiftUI. I do think it's something we should figure out now, just in case we need to bail on this approach.

Reply:
@tonisevener Yeah 😕 that's part of what I'm still unsatisfied by here with the button. Dismissing the context menu triggers the diff cell tap gesture unless the user's dismiss gesture is a swipe on list. I'm still puzzled by it – if you have any spark of insight on it I'm all ears but no worries if not. I've got a tiny PR or 2 to get up first, then I'll swing back around and work on this problem again.

Reply:
No problem - I'm good with merging this as-is if you are, just wanted to make sure you were aware of it. Thanks!